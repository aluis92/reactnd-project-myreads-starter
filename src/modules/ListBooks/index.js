import React from 'react';
import {
    node,
    string,
} from 'prop-types';
import BookListTitle from '../../elements/BookListTitle';
import OpenSearch from '../../elements/OpenSearch';

import {
    Container,
    Content,
} from './styles';

function ListBooks({
    title,
    children,
}) {
    return (
        <Container>
            <Content>
                <BookListTitle>{title}</BookListTitle>
                <div>
                    {children}
                </div>
            </Content>

            <OpenSearch
                to="/search"
                text="Add a book"
            />
        </Container>
    );
}

ListBooks.defaultProps = {
    children: null,
    title: '',
};

ListBooks.propTypes = {
    children: node,
    title: string,
};

export default ListBooks;
