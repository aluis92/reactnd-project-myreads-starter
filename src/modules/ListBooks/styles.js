import styled from 'styled-components';

export const Container = styled.section`
    position: relative
`;

export const Content = styled.div`
    padding: 0 0 80px;
    flex: 1;
`;
