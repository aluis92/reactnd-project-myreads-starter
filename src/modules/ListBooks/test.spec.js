import React from 'react';
import {
    shallow,
} from 'enzyme';

import ListBooks from './index';

describe('ListBooks component', () => {
    it('should render without crashing', () => {
        expect(shallow(
            <ListBooks />,
        )).toMatchSnapshot();
    });
});
