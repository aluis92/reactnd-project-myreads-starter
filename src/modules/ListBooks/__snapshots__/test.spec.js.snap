// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBooks component should render without crashing 1`] = `
<styled.section>
  <styled.div>
    <BookListTitle />
    <div />
  </styled.div>
  <OpenSearch
    text="Add a book"
    to="/search"
  />
</styled.section>
`;
